sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultDataAction.java

```diff
@@ -0,0 +1,182 @@
+    public static final KeyVaultDataAction IMPORT_HSM_KEY = fromString("Microsoft.KeyVault/managedHsm/keys/import/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/keys/deletedKeys/delete for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction PURGE_DELETED_HSM_KEY =
+        fromString("Microsoft.KeyVault/managedHsm/keys/deletedKeys/delete");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/securitydomain/download/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction DOWNLOAD_HSM_SECURITY_DOMAIN =
+        fromString("Microsoft.KeyVault/managedHsm/securitydomain/download/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/securitydomain/upload/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction UPLOAD_HSM_SECURITY_DOMAIN =
+        fromString("Microsoft.KeyVault/managedHsm/securitydomain/upload/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/securitydomain/upload/read for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction READ_HSM_SECURITY_DOMAIN_STATUS =
+        fromString("Microsoft.KeyVault/managedHsm/securitydomain/upload/read");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/securitydomain/transferkey/read for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction READ_HSM_SECURITY_DOMAIN_TRANSFER_KEY =
+        fromString("Microsoft.KeyVault/managedHsm/securitydomain/transferkey/read");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/backup/start/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction START_HSM_BACKUP = fromString("Microsoft.KeyVault/managedHsm/backup/start/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/restore/start/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction START_HSM_RESTORE = fromString("Microsoft.KeyVault/managedHsm/restore/start/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/backup/status/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction READ_HSM_BACKUP_STATUS =
+        fromString("Microsoft.KeyVault/managedHsm/backup/status/action");
+
+    /**
+     * Static value Microsoft.KeyVault/managedHsm/restore/status/action for {@link KeyVaultDataAction}.
+     */
+    public static final KeyVaultDataAction READ_HSM_RESTORE_STATUS =
+        fromString("Microsoft.KeyVault/managedHsm/restore/status/action");
+
+    /**
+     * Creates or finds a {@link KeyVaultDataAction} from its string representation.
+     *
+     * @param name A name to look for.
+     *
+     * @return The corresponding {@link KeyVaultDataAction}.
+     */
+    @JsonCreator
+    public static KeyVaultDataAction fromString(String name) {
+        return fromString(name, KeyVaultDataAction.class);
+    }
+}
```

Comment:
These all look correct, but unsorted it's harder to compare with APIView for .NET's. Take a look at https://apiview.dev/Assemblies/Review/be28e18d85544b27b7ab638cc2b87880#Azure.Security.KeyVault.Administration.Models.KeyVaultDataAction when you have a new upload in APIView available and compare.